cpp/log/cert.cc

```diff
@@ -355,22 +358,36 @@ Cert::Status Cert::IsSignedBy(const Cert& issuer) const {
 
   int ret = X509_verify(x509_, issuer_key);
   EVP_PKEY_free(issuer_key);
-  if (ret < 0) {
-    unsigned long err = ERR_peek_last_error();
-    int reason = ERR_GET_REASON(err);
-    if (ERR_GET_LIB(err) == ERR_LIB_ASN1 &&
+  if (ret == 1) {
+    return TRUE;
+  }
+  unsigned long err = ERR_peek_last_error();
+  if (!err) {
+    return FALSE;
+  }
+  int reason = ERR_GET_REASON(err);
```

Comment:
Could you make `err`, `reason`, `lib`, & `unsupported_algorithm` all be `const` while you're here?